tests/mocha/connection_checker_test.js

```diff
@@ -341,4 +341,117 @@ suite('Connection checker', function() {
       chai.assert.isFalse(this.checker.doTypeChecks(this.con1, this.con2));
     });
   });
+  suite('Dragging Checks', function() {
+    suite('Stacks', function() {
+      setup(function() {
+        this.workspace = Blockly.inject('blocklyDiv');
+        // Load in three blocks: A and B are connected (next/prev); B is unmovable.
+        Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(`<xml xmlns="https://developers.google.com/blockly/xml">
+        <block type="text_print" id="wlc#l2D-,6mSLKY=E$VL" x="-76" y="-112">
+          <next>
+            <block type="text_print" id="T9cdR}mwDn3Xl!*g|mw8" movable="false">
+              <comment pinned="true" h="32" w="88.5333251953125">Immovable</comment>
+            </block>
+          </next>
+        </block>
+        <block type="text_print" id="ZgFlirwB{~}%l/BZ+j?R" x="47" y="-118"/>
+      </xml>`), this.workspace);
```

Comment:
Optional cleanup: This is kind of difficult to read with the extra properties (i.e. ids, coordinates, etc).  Could also be converted to JSON if you feel so inclined.

Here & below

Reply:
Cleaned it up! Made the IDs A, B, and C and removed the comment.

I left the coordinates only because while debugging I found it helpful to copy this into the playground, and was confused by the overlapping blocks if there are no coordinates lol.